### ISSUE TITLE
RxJava 2.0 Planning?

### ISSUE THREAD

**Issue by thomasnield** [davidmoten/rxjava-jdbc#63]
Hey @davidmoten,

This is probably going to be a somewhat large refactoring task. What are your plans with RxJava 2.0 being on the horizon? Will there be two versions of this codebase on two branches? With two separate 1.0.x and 2.0.x releases?

Tom

**Comment by davidmoten** [comment]
Hi Tom, haven't planned it yet. I'd support both rxjava 1.x and 2.x for a
good while but not sure if I'd branch this repo or fork to new repo
rxjava-jdbc2.

RxJava2 is a way off production ready and lots of api change still
happening so no rush I think.

Cheers
Dave

**Comment by thomasnield** [comment]
Well that's a relief,  haha I was already planning my own libraries for the transition. I'm glad there is good reason to hold off then. 

Sounds good. Ill adopt that fork approach for my stuff and recommend it to others, that's a good idea. You can close this issue or leave it open.

**Comment by akarnokd** [comment]
Switching between branches is somewhat inconvenient for me in RxJava. I'd prefer separate project so you can have both of them open in the same Eclipse workspace (not sure how IntelliJ would be affected).

I may be able to help you with porting this to 2.x but it really depends on the extra features you did or how deeply you depended on 1.x structure that has to be remade.

**Comment by thomasnield** [comment]
@akarnokd I'm glad you're here, thanks for helping out. I can try to help you and @davidmoten but my low-level RxJava knowledge is limited, and I'm a bit overloaded with projects right now. I can assist with testing functionality on my end against SQLite, PostgreSQL, and MySQL.

**Comment by davidmoten** [comment]
thanks @akarnokd. Porting shouldn't be a problem for this one, thanks for the offer of help!

Speaking of switching branches between RxJava 1.x and 2.x in Eclipse what I did was cloned RxJava again to an RxJava2 folder and imported that to Eclipse as project rxjava2. That way I can have them both open.

**Comment by akarnokd** [comment]
Just fire up a new repo or branch (build ready) and I'll do the conversion.

**Comment by akarnokd** [comment]
Looks like rxjava-extras needs the conversion first, at least those parts used by this library.

**Comment by davidmoten** [comment]
Yep, I'll have a look at them both (or just the bits that rxjava-jdbc uses)
and plan it. rxjava-jdbc needs a rewrite I reckon, it predates useful
rxjava operators and predates my improved knowledge. Might leave that till
after a simple migration to 2.x though.

**Comment by krishna81m** [comment]
Hello, just checking if 2.x support is available in beta?

**Comment by thomasnield** [comment]
@krishna81m I don't think it is. Last time I checked [RxJava-Extras](https://github.com/davidmoten/rxjava-extras) needed to be ported over to RxJava2 since it is a dependency. There likely needs to be some design decisions regarding Flowables vs Observables, and whether to support one or both (I'm leaning towards the latter since both can be included in the builder). 

If you need to urgently use this with RxJava 2.0, I recommend looking at [RxJava2Interop](https://github.com/akarnokd/RxJava2Interop) by David Karnok to convert the Observables into its RxJava 2.0 counterparts. 

It's a bit unfortunate David and the other RxJava gurus are ridiculously busy. I'd take a stab at it but I'm busy writing the [_Learning RxJava_ book with Packt](https://www.packtpub.com/application-development/learning-rxjava). If anybody can provide help I know many people (including myself) would be highly appreciative.

**Comment by davidmoten** [comment]
>Hello, just checking if 2.x support is available in beta?
Thanks  @thomasnield for answering. Use RxJava2Interop as suggested to use the existing library. 

Work is under way on rxjava2-jdbc at https://github.com/davidmoten/rxjava2-jdbc as a complete rewrite. I'll get back to it shortly. New version is discussed at https://github.com/davidmoten/rxjava2-jdbc/wiki but is still in development (not passing all tests at the moment).

**Comment by ArekCzarnik** [comment]
any news on rxjava2-jdbc? it is still under development ?

**Comment by thomasnield** [comment]
@ArekCzarnik David Moten is pretty busy. I've created my own implementation in Kotlin to get by in the interim.

**Comment by davidmoten** [comment]
I imagine I'll juggle my priorities to favour rxjava2-jdbc given enough interest. It's on my radar (as is your book @thomasnield, thanks for answering).

**Comment by davidmoten** [comment]
Just got past a blocker in rxjava2-jdbc, development can accelerate now, if you watch the rxjava2-jdbc project you'll probably notice some action in the next week or so.

**Comment by thomasnield** [comment]
@davidmoten awesome, thanks David!

**Comment by davidmoten** [comment]
An update on *rxjava2-jdbc*:

* I'm happy with its current state but I'll seek to increase test coverage over the next couple of weeks and will make minor api changes
* I haven't added CallableStatement support yet 
* not planning on supporting `autoMap` to constructors (just to interfaces)
* I'll release an RC in a couple of weeks

You're welcome to have a play with it, I'll endeavour to keep master buildable till release.

**Comment by davidmoten** [comment]
Ah, needs more work, don't bother playing with it yet!

**Comment by davidmoten** [comment]
Ok, rxjava2-jdbc is up and running, tests pass now, 0.1-RC1 has been released to Maven Central so anyone interested have a play! https://github.com/davidmoten/rxjava2-jdbc

**Comment by thomasnield** [comment]
@davidmoten awesome, thanks David! I'm going to share this right now.
